src/ansys/systemcoupling/core/charts/chart_datatypes.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Optional
+
+# Common intermediate metadata and data types for chart series information.
+# All series are assumed to belong to an interface and a data transfer within the interface
+
+# Idea is that data should be processed into this format from whatever raw source (CSV file,
+# streamed data, etc.) and the actual charts will be built from this.
+
+
+class SeriesType(Enum):
+    CONVERGENCE = 1
+    SUM = 2
+    WEIGHTED_AVERAGE = 3
+
+
+@dataclass
+class TransferSeriesInfo:
+    """Information about chart series data associated with a data transfer.
+
+    The `data_index` fields are used by the data source processor to associate
+    this information (likely obtained from heading or other metadata) with the
+    correct data series. It indexes into the full list of series associated
+    with a given interface.
```

Comment:
```suggestion
    The ``data_index`` fields are used by the data source processor to associate
    this information (likely obtained from heading or other metadata) with the
    correct data series. It indexes into the full list of series associated
    with a given interface.
```
It isn't clear if the pronoun "it" is referring to "this information" or to "``data_index`` fields. I assume the first given the verb (indexes), but using a noun rather than a pronoun would provide clarity.

Reply:
I was referring to `data_index`. I think this is confusing because I switched from plural ("`data_index` fields") to singular ("it").

"Fields" is correct in this context. It is standard teminology for the items of data in a `dataclass`.

BTW this was not intended to be complete or polished documentation at this stage. A lot of it consists of sketchy notes for my own benefit while this work is in progress. Generally, I would recommend not paying too much attention to PRs that are marked as draft.